server/src-lib/Hasura/Backends/Postgres/DDL/RunSQL.hs

```diff
@@ -300,7 +300,7 @@ runTxWithMetadataCheck source sourceConfig txAccess tableCache functionCache cas
         -- Before running the @'tx', fetch metadata of existing tables and functions from Postgres.
         let tableNames = M.keys tableCache
             computedFieldFunctions = concatMap getComputedFieldFunctions (M.elems tableCache)
```

Comment:
Not sure if it makes sense to change `getComputedFieldFunctions` to return `HashSet` instead of list. There doesn't seem to be an equivalent of `concatMap`/`concat` for `HashSet` (coming from Scala, it is pretty frustrating that the Haskell containers don't implement common interfaces and are not interoperable).